### PR TITLE
feat: drop rusha

### PR DESCRIPTION
**Pull request by ThaUnknown**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[X] Bug fix
[X] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
REQUIRES ESM

Dropped `simple-sha1` for `uint8-util` in order to drop `rusha`. `uint8-util` uses `node:crypto` in node and Web Crypto API in browser, as `node:crypto`'s sha1 is an order of magnitude faster than the Web Crypto API.

This also exposes a few snippets from [wt-common](https://github.com/jimmywarting/wt-common/) for the future removal of `Buffer.from` and `Buffer.toString`, however not all of them, as some plain JS are a LOT faster `Buffer.from`, and wt-common forced `Buffer.from` if node was used.

*For now* `uint8-util` is 85% smaller than `simple-sha1`, but that will change to roughly 65% smaller once I get around to adding all the Buffer.from-like functionality to uint8.
**Which issue (if any) does this pull request address?**
https://github.com/webtorrent/webtorrent/issues/2217
https://github.com/orgs/webtorrent/projects/1#card-60702974
**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

**Comment by DiegoRBaquero** [comment]
Given this requires ESM, it should target v2

**Comment by ThaUnknown** [comment]
> Given this requires ESM, it should target v2

probably, but realistically we will only be able to merge this after v2 is merged into master

**Review by DiegoRBaquero — APPROVED** [review]

**Pull request merged**
